cross_list_courses/static/cross_list_courses/js/controllers/ListController.js

```diff
@@ -110,14 +121,87 @@
                 }).finally(function cleanupAfterDelete() {
                     // always reload, in case the failure was a 404
                     $scope.dtInstance.reloadData();
-                    $scope.operationInProgress = null;
                 });
 
             return promise;
         };
+        $scope.search = function() {
+            $scope.clearMessages();
+            $scope.setOperationInProgress('search');
+            // Call within timeout to prevent
+            // https://docs.angularjs.org/error/$rootScope/inprog?p0=$apply
+            $timeout(function () {
+                $scope.dtInstance.reloadData();
+            }, 0);
+        };
+        $scope.setOperationInProgress = function(newSetting) {
+            // disable datatable and show progress bar if newSetting is truthy
+            $timeout(function() {
+                // notify UI to start/stop showing in-progress messaging
+                $scope.operationInProgress = newSetting;
+                // enable/disable interactive data table elements
+                $scope.setDataTableInteraction(!newSetting);
```

Comment:
(to be more concrete, if the `newSetting` string is e.g. `'add'`, then `newSetting` is truthy, and `!newSetting` is `false`, which disables the datatable interaction).